ClosedXML/Excel/PivotTables/XLPivotTable.cs

```diff
@@ -27,7 +27,39 @@ public XLPivotTable(IXLWorksheet worksheet)
         }
 
         public IXLCell TargetCell { get; set; }
-        public IXLRange SourceRange { get; set; }
+
+        private IXLRange sourceRange;
+
+        public IXLRange SourceRange
+        {
+            get { return sourceRange; }
+            set
+            {
+                if (value is IXLTable)
+                    SourceType = XLPivotTableSourceType.Table;
+                else
+                    SourceType = XLPivotTableSourceType.Range;
+
+                sourceRange = value;
+            }
+        }
+
+        public IXLTable SourceTable
+        {
+            get
+            {
+                if (this.SourceType == XLPivotTableSourceType.Table)
+                    return SourceRange as IXLTable;
+                else
+                    throw new InvalidOperationException("This pivot table's source is not a table.");
```

Comment:
I would prefer `return null` but do not insist.

Reply:
Done.